pkg/api/internalclient/sshkey.go

```diff
@@ -0,0 +1,70 @@
+package internalclient
+
+import (
+	"fmt"
+
+	"github.com/shellhub-io/shellhub/pkg/models"
+)
+
+// sshkeyAPI defines methods for interacting with SSH key-related functionality.
+type sshkeyAPI interface {
+	// GetPublicKey retrieves the public key identified by the provided fingerprint and tenant.
+	GetPublicKey(fingerprint, tenant string) (*models.PublicKey, error)
+
+	// CreatePrivateKey creates a new private key.
+	CreatePrivateKey() (*models.PrivateKey, error)
+
+	// EvaluateKey evaluates whether a given public key identified by fingerprint is valid for a device and username combination.
+	EvaluateKey(fingerprint string, dev *models.Device, username string) (bool, error)
+}
+
+func (c *client) GetPublicKey(fingerprint, tenant string) (*models.PublicKey, error) {
+	pubKey := new(models.PublicKey)
+
+	resp, err := c.http.
+		R().
+		SetResult(&pubKey).
+		Get(fmt.Sprintf("/internal/sshkeys/public-keys/%s/%s", fingerprint, tenant))
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode() == 404 {
+		return nil, ErrNotFound
+	}
```

Comment:
Are your idea standardizing the errors from the internal client?

We have something like this too.

https://github.com/shellhub-io/shellhub/blob/master/pkg/api/client/errors.go#L40